rust/docs/async/Trait_ThingTypeAPI.adoc

```diff
@@ -0,0 +1,695 @@
+[#_trait_ThingTypeAPI]
+=== Trait ThingTypeAPI
+
+// tag::methods[]
+[#_trait_ThingTypeAPI_tymethod_label]
+==== label
+
```

Comment:
Can we have a list of implementors here?

Reply:
Yes, we can.